xcp_d/utils/dcan2fmriprep.py

```diff
@@ -11,315 +12,397 @@
+        # NOTE: We're using the T1w image as a transform. This doesn't make sense.
+        t1w_to_template_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5",
         )
-        regfile2 = (
-            anatdir + sub_id + "_" + sess + "_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5"
+        copy_dictionary[t1w_orig].append(t1w_to_template_fmriprep)
+
+        # NOTE: We're using the T1w image as a transform. This doesn't make sense.
+        template_to_t1w_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5",
         )
+        copy_dictionary[t1w_orig].append(template_to_t1w_fmriprep)
 
-        lMid = anatdir + sub_id + "_" + sess + "_hemi-L_midthickness.surf.gii"
-        rMid = anatdir + sub_id + "_" + sess + "_hemi-R_midthickness.surf.gii"
-
-        lpial = anatdir + sub_id + "_" + sess + "_hemi-L_pial.surf.gii"
-        rpial = anatdir + sub_id + "_" + sess + "_hemi-R_pial.surf.gii"
-
-        lwhite = anatdir + sub_id + "_" + sess + "_hemi-L_smoothwm.surf.gii"
-        rwhite = anatdir + sub_id + "_" + sess + "_hemi-R_smoothwm.surf.gii"
-
-        linf = anatdir + sub_id + "_" + sess + "_hemi-L_inflated.surf.gii"
-        rinf = anatdir + sub_id + "_" + sess + "_hemi-R_inflated.surf.gii"
-
-        newanatfiles = [
-            t1wim,
-            t1seg,
-            t1ribbon,
-            t1brainm,
-            regfile1,
-            regfile2,
-            lMid,
-            rMid,
-            lpial,
-            rpial,
-            lwhite,
-            rwhite,
-            linf,
-            rinf,
-        ]
+        brainmask_orig = os.path.join(anat_dir_orig, "brainmask_fs.nii.gz")
+        brainmask_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_desc-brain_mask.nii.gz",
+        )
+        copy_dictionary[brainmask_orig] = [brainmask_fmriprep]
 
-        for i, j in zip(dcanimages, newanatfiles):
-            copyfileobj_example(i, j)
+        ribbon_orig = os.path.join(anat_dir_orig, "ribbon_orig.nii.gz")
+        ribbon_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_desc-ribbon_orig.nii.gz",
+        )
+        copy_dictionary[ribbon_orig] = [ribbon_fmriprep]
 
-        # get masks and transforms
+        dseg_orig = os.path.join(anat_dir_orig, "aparc+aseg.nii.gz")
+        dseg_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_{ses_id}_dseg.nii.gz")
+        copy_dictionary[dseg_orig] = [dseg_fmriprep]
 
-        wmmask = glob.glob(anat_dirx + "/wm_2mm_*_mask_eroded.nii.gz")[0]
-        csfmask = glob.glob(anat_dirx + "/vent_2mm_*_mask_eroded.nii.gz")[0]
-        tw1tonative = wmmask
-
-        # get task and idx  run 01
-        func_dirx = dcan_dir + "/" + sub_id + "/ses-" + ses_id[0] + "/files/MNINonLinear/Results/"
```

Comment:
AFAICT, this means that the script looped over all sessions, but copied functional files from only the first session's DCAN folder into _all_ of the sessions' fMRIPrep folders.